src/rgw/rgw_rados.cc

```diff
@@ -8957,14 +8957,14 @@ int RGWRados::get_target_shard_id(const RGWBucketInfo& bucket_info, const string
                                   int *shard_id)
 {
   int r = 0;
-  switch (bucket_info.bucket_index_shard_hash_type) {
-    case RGWBucketInfo::MOD:
-      if (!bucket_info.num_shards) {
+  switch (bucket_info.layout.current_index.layout.normal.hash_type) {
```

Comment:
can this function take `const bucket_index_normal_layout& layout` instead of `const RGWBucketInfo& bucket_info`?